libqtile/layout/base.py

```diff
@@ -293,19 +293,34 @@ def focus_previous(self, win):
             return self[idx - 1]
         return None
 
-    def add(self, client, offset_to_current=0):
-        """
-        Insert the given client into collection at position of the current.
-
-        Use parameter 'offset_to_current' to specify where the client shall be
-        inserted. Defaults to zero, which means at position of current client.
-        Positive values are after the client.
-        """
-        pos = max(0, self._current_idx + offset_to_current)
-        if pos < len(self.clients):
-            self.clients.insert(pos, client)
+    def add(self, client, offset_to_current=0, client_position=None):
+        """
+        If client_position is set by user, insert the given client depending of
+          its value : top, bottom, after_current, before_current.
+        Else:
+          Insert the given client into collection at position of the current.
+
+          Use parameter 'offset_to_current' to specify where the client shall be
+          inserted. Defaults to zero, which means at position of current client.
+          Positive values are after the client.
+        """
+        if client_position is not None:
+            if client_position == "after_current":
+                pos = max(0, self._current_idx + 1)
+                self.clients.insert(pos, client)
+            elif client_position == "before_current":
+                pos = max(0, self._current_idx)
+                self.clients.insert(pos, client)
+            elif client_position == "top":
+                self.append_head(client)
+            else:  # ie client_position == "bottom"
+                self.append(client)
         else:
-            self.clients.append(client)
+            pos = max(0, self._current_idx + offset_to_current)
+            if pos < len(self.clients):
+                self.clients.insert(pos, client)
+            else:
+                self.clients.append(client)
         self.current_client = client
```

Comment:
May I suggest something like this?

```python
        if client_position == "after_current":
            return self.add(client, offset_to_current=1)
        if client_position == "before_current":
            return self.add(client, offset_to_current=0)
        if client_position == "top":
            return self.append_head(client)
        if client_position == "bottom":
            return self.append(client)
            
        pos = max(0, self._current_idx + offset_to_current)
        ...
```

It makes the diff simpler and reuse the existing `offset_to_current`, which I think is good to ensure we don't break compatibility in the future. What do you think?

Reply:
Nah, that won't work; `append` and `append_head` do not set the current client. :(